### ISSUE TITLE
Product Collection - enable automatic migration

### ISSUE THREAD

**Issue by kmanijak** [woocommerce/woocommerce-blocks#10431]
After allowing Product Collection block to be used (https://github.com/woocommerce/woocommerce-blocks/issues/10430), we should enable automatic migration from Products (Beta) block to Product Collection block couple of WC Blocks versions later.

That includes:
- enable automatic migration [here](https://github.com/woocommerce/woocommerce-blocks/pull/10267/files#diff-e9517b2fd3bcc6d45a42d1f2eea15b221e8d7285a41c4f3df6fa2eb8db2de3acR6)
- if migration was not reverted, don't include Products (Beta) block in the inserter
- replace Products (Beta) block with Product Collection in the patterns (or remove the Products (Beta) patterns) - to be decided

**Comment by Marc-pi** [comment]
yep, that would be cool. For now, we have to edit our templates each by each
but there's a need to enhance the block's options, there are few at this time

**Comment by kmanijak** [comment]
> yep, that would be cool.

Thanks for the feedback @Marc-pi! The idea is to gather some early feedback about Product Collection block before enabling auto migration.

> but there's a need to enhance the block's options, there are few at this time

Do you mind providing more details? Do you mean Product Collection block? What enhancements would you like to see? Your feedback is valuable to us so we can make the block better! :)

cc: @imanish003 for visibility 🙌

**Comment by Marc-pi** [comment]
yep, i added this one yesterday https://github.com/woocommerce/woocommerce-blocks/issues/10983 (add product type as option)
and playing yesterday with filter blocks and product collection, i stuck to this need https://github.com/woocommerce/woocommerce-blocks/issues/10985#issuecomment-1722872497 aka dynamic filter's (new Collection Filter block used in conjunction with the Product Collection query block)
